backend/util/file_io.py

```diff
@@ -142,6 +147,49 @@ def get_blake2_str(
         self.logger.error("Failed to hash file")
         raise Exception("ErrorHashingFile")  # I am really sorry about this
 
+    def generate_local_filepath_from_hash(
```

Comment:
These should be on main, not sure why they are, somewhat confused, will investigate further.